Assets/MixedRealityToolkit.Services/TeleportSystem/MixedRealityTeleportSystem.cs

```diff
@@ -283,11 +267,18 @@ private void ProcessTeleportationRequest(TeleportEventData eventData)
             }
 
             float height = targetPosition.y;
-            targetPosition -= CameraCache.Main.transform.position - cameraParent.position;
+            targetPosition -= CameraCache.Main.transform.position - MixedRealityPlayspace.Position;
             targetPosition.y = height;
-            cameraParent.position = targetPosition;
 
-            cameraParent.RotateAround(CameraCache.Main.transform.position, Vector3.up, targetRotation.y - CameraCache.Main.transform.eulerAngles.y);
+            MixedRealityPlayspace.PerformTransformation(
+                p =>
+                {
+                    p.position = targetPosition;
+                    p.RotateAround(
+                        CameraCache.Main.transform.position, 
+                        Vector3.up, 
+                        targetRotation.y - CameraCache.Main.transform.eulerAngles.y);
+                });
```

Comment:
Same comment from the other section - if this is going to be done in any sort of inner loop it would be best to avoid lambdas (which may mean we have to change the function definition of PerformanceTransformation